### PR TITLE
Add explicit_top_level_acl opt-in rule

### LINKED ISSUE #58: Validate explicit ACL modifiers for top-level declarations
Top-level declarations should use explicit ACL keywords (`public`, `internal`, `private`).

### PR DESCRIPTION
**Pull request by marcelofabri**
Continued from #1473

Fixes #58

@masters3d I also took the liberty of renaming the rule so hopefully it's clearer what it enforces (the fact that it only checks for `internal` is an implementation detail).

This will probably have different results from #1473 since it validates all declarations (not only types declarations) - I think this is more aligned with @jpsim's original intention in #58.

### CODE REVIEW

**Comment by jpsim** [comment]
How about naming this rule `ExplicitTopLevelACL`? `ExplicitACLInTopLevelDeclaration` is a bit wordy.

**Comment by marcelofabri** [comment]
> How about naming this rule `ExplicitTopLevelACL`? `ExplicitACLInTopLevelDeclaration` is a bit wordy.

Sounds good, will rename and push it.

File: Source/SwiftLintFramework/Rules/ExplicitACLInTopLevelDeclarationRule.swift [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+public struct ExplicitACLInTopLevelDeclarationRule: OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "explicit_acl_in_top_level_declaration",
+        name: "Explicit ACL in Top Level Declaration",
+        description: "Top-level declarations should specify Access Control Labels explicitly.",
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/ExplicitACLInTopLevelDeclarationRule.swift]: "ACL" stands for "Access Control Level", so perhaps reword to "Top-level declarations should specify Access Control Level keywords explicitly."

**Comment by jpsim** [comment]
🎉
